config/fcos/v1_6_exp/translate.go

```diff
@@ -133,6 +135,13 @@ func (c Config) processBootDevice(config *types.Config, ts *translate.Translatio
 		wantEFIPart = true
 	case *layout == "ppc64le":
 		wantPRePPart = true
+	case *layout == "s390x-eckd":
```

Comment:
Since s390x-zfcp is the same as s390x-eckd, we could have both cases fall into the same logic 
IE
```
case x:
case y:
       dowork = true
```

Reply:
Yes  logically it does not do anything with two different layout with one input.
But the plan was user to force use /dev/dasd[a-z] for eckd and /dev/sd[a-z] so that during the layout preference it set appropriate root device; let say for dasd `/dev/dasda2`. But , I was not able to achieve that under translate.go. I tried with regexp. Somehow it was crashing with memory error. I thought once created the PR will add the device arch once it is discuss within PR.. That's why made it open.

1, I imported regexp.
2. then did `diskRe = regexp.MustCompile("(/dev/dasd[a-z]$|/dev/sd[a-z]$)")
3. diskRe.FindString(*c.BootDevice.Luks.Device)
 while validating with appropriate layouts getting panic on memory error

Reply:
Can we use the following conditions?

In translate.go will have one logic as you corrected 
```

case s390x-zfcp:
case s390x-eckd:
         wantS390X = true


```

 below condition in validate.go  to fix , so that user force to use either dasd[a-z] or sd[a-z].  If not then it give an error.

```
var dasdRe = regexp.MustCompile("(/dev/dasd[a-z]$)")
var sdRe = regexp.MustCompile("(/dev/sd[a-z]$)")

if d.Layout != nil && (*d.Layout == "s390x-zfcp" ||  *d.Layout == "s390x-eckd") {
		if util.NilOrEmpty(d.Luks.Device) {
			r.AddOnError(c.Append(*d.Layout), common.ErrNoLuksBootDevice)
		}
		if len(d.Mirror.Devices) > 0 {
			r.AddOnError(c.Append(*d.Layout), common.ErrMirrorNotSupport)
		}
		if *d.Layout == "s390x-zfcp" && util.NotEmpty(d.Luks.Device) && !sdRe.MatchString(*d.Luks.Device) {
			r.AddOnError(c.Append(*d.Layout), common.ErrNoLuksBootDevice)
		}
		if *d.Layout == "s390x-eckd" && util.NotEmpty(d.Luks.Device) && !dasdRe.MatchString(*d.Luks.Device) {
			r.AddOnError(c.Append(*d.Layout), common.ErrNoLuksBootDevice)
		}
	}
```